packages/modeling/src/maths/utils/intersect.js

```diff
@@ -6,10 +6,11 @@
  * @param {vec2} p2 - second point of first line segment
  * @param {vec2} p3 - first point of second line segment
  * @param {vec2} p4 - second point of second line segment
+ * @param {Boolean} noEndpointTouch - skip intersections at segment endpoints
  * @returns {vec2} intersection point of the two line segments, or undefined
  * @alias module:modeling/maths/utils.intersect
  */
-export const intersect = (p1, p2, p3, p4) => {
+export const intersect = (p1, p2, p3, p4, noEndpointTouch) => {
```

Comment:
I'm not sure which is better... another parameter or another function.

If parameter then the default should be the previous behavior; includeEndPoints = false

If functions then there could be 'intersectWithin()' and 'intersect()', or something like this.

By the way, i think the original description is wrong. '..., end points **excluded**' is correct.

Reply:
The previous behavior is to INCLUDE endpoints. That is why I made the parameter `noEndpointTouch` because the default previous behavior was YES endpoint touch. So the original description was right.

I can imagine a few ways to approach this:

1. `noEndpointTouch = false` as the default. This is essentially how I made this PR (since the default is actually undefined). I could add the default `= false` though.
2. `endpointTouch = true` as the default. This is fine, just inverted from `segment_intersection` in martinez.
3. Split into `intersect` and `intersectInside` or some other variation. It would be nice if we can re-use the same code though for both.

I'm personally fine with any of the options. I think my preference is for option 2 or option 1. But let me know what you prefer and I'll change it. They are all fine to me.

Reply:
Ok. Let's go with option #215

Reply:
Ok I updated to use the `endpointTouch` optional parameter. Let me know what you think.

Reply:
@platypii Did we update the typescript definitions to accept the extra boolean?

Reply:
Good catch, I missed that. I will update in a new PR. Thanks @youssefsharief